pubsub/pubsub.go

```diff
@@ -40,3 +56,60 @@ func Ping(pubsub PubSub) error {
 		return fmt.Errorf("ping is not implemented by this pubsub")
 	}
 }
+
+// DefaultBulkMessager is default implemnetation for BukMessager
+type DefaultBulkMessager struct {
+	p PubSub
+}
+
+// NewDefaultBulkMessager to create new DefaultBulkMessager for a PubSub
+func NewDefaultBulkMessager(pubsub PubSub) DefaultBulkMessager {
+	return DefaultBulkMessager{
+		p: pubsub,
+	}
+}
+
+// BulkPublish publishes messages to a broker that does not support bulk publish.
+// If 'bulkPublishKeepOrder' metadata is set to true, the messages are published serially
+// in the same order, otherwise they are published in parallel.
+func (p *DefaultBulkMessager) BulkPublish(req *BulkPublishRequest) (BulkPublishResponse, error) {
+	if req.Metadata[bulkPublishKeepOrderKey] == "true" {
+		return p.bulkPublishSerial(req)
```

Comment:
What is this new flag? 

Currently how is in order publishing supported?

Reply:
@shubham1172 my concern here is that this might confuse the user and provide a false notion that in order publishing and processing is guaranteed whereas that is only available if broker supports it like for example partition key in AZ events hub..  

@berndverst @yaron2 @artursouza i think this flag should either be renamed or removed.... Thoughts?

Reply:
The intent was to allow a fast way to publish everything in parallel, and a slower way to publish serially (and in-order if the broker supports). We can rename this to `bulkPublishSerially` if that helps to remove the confusion.